src/Analysis/Ast/Test/PathClassificationTests.cs

```diff
@@ -0,0 +1,189 @@
+                new PythonLibraryPath(venvLib, PythonLibraryPathType.StdLib),
+                new PythonLibraryPath(venv, PythonLibraryPathType.StdLib),
+                new PythonLibraryPath(venvSitePackages, PythonLibraryPathType.Site),
+            });
+
+            userPaths.Should().BeEmpty();
+        }
+
+        [TestMethod]
+        public void WithSrcDir() {
+            var appPath = TestData.GetTestSpecificPath("app.py");
+            var root = Path.GetDirectoryName(appPath);
+
+            var venv = Path.Combine(root, "venv");
+            var venvLib = Path.Combine(venv, "Lib");
+            var venvSitePackages = Path.Combine(venvLib, "site-packages");
+
+            var src = Path.Combine(root, "src");
+
+            var fromInterpreter = new[] {
+                new PythonLibraryPath(venvLib, PythonLibraryPathType.StdLib),
+                new PythonLibraryPath(venv, PythonLibraryPathType.StdLib),
+                new PythonLibraryPath(venvSitePackages, PythonLibraryPathType.Site),
+            };
+
+            var fromUser = new[] {
+                "./src",
+            };
+
+            var (interpreterPaths, userPaths) = PythonLibraryPath.ClassifyPaths(root, _fs, fromInterpreter, fromUser);
+
+            interpreterPaths.Should().BeEquivalentToWithStrictOrdering(new[] {
+                new PythonLibraryPath(venvLib, PythonLibraryPathType.StdLib),
+                new PythonLibraryPath(venv, PythonLibraryPathType.StdLib),
+                new PythonLibraryPath(venvSitePackages, PythonLibraryPathType.Site),
+            });
+
+            userPaths.Should().BeEquivalentToWithStrictOrdering(new[] {
+                new PythonLibraryPath(src, PythonLibraryPathType.Unspecified),
+            });
+        }
+
+        [TestMethod]
+        public void NormalizeUser() {
+            var appPath = TestData.GetTestSpecificPath("app.py");
+            var root = Path.GetDirectoryName(appPath);
+
+            var src = Path.Combine(root, "src");
+
+            var fromUser = new[] {
+                "./src/",
+            };
+
+            var (interpreterPaths, userPaths) = PythonLibraryPath.ClassifyPaths(root, _fs, Array.Empty<PythonLibraryPath>(), fromUser);
+
+            interpreterPaths.Should().BeEmpty();
+
+            userPaths.Should().BeEquivalentToWithStrictOrdering(new[] {
+                new PythonLibraryPath(src, PythonLibraryPathType.Unspecified),
+            });
+        }
+
+        [TestMethod]
+        public void NestedUser() {
+            var appPath = TestData.GetTestSpecificPath("app.py");
+            var root = Path.GetDirectoryName(appPath);
+
+            var src = Path.Combine(root, "src");
+            var srcSomething = Path.Combine(src, "something");
+            var srcFoo = Path.Combine(src, "foo");
+            var srcFooBar = Path.Combine(srcFoo, "bar");
+
+            var fromUser = new[] {
+                "./src",
+                "./src/something",
+                "./src/foo/bar",
```

Comment:
Do we need to have another test that will have there paths in a different order?

Reply:
Yes, I should add one; I had one but deleted it when I undid the depth sorting.

Reply:
And do we need depth sorting after all?

Reply:
The problem with the depth sorting is that it doesn't seem to fix things quite yet; forcing a sorting would mean that a user can't specify a certain ordering (which I think is related to the issue where a module is only allowed to have a single name, versus being importable from any of its names).

Reply:
Got it